cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -0,0 +1,82 @@
+import re
+from typing import List, Optional, Tuple, Union
+import pandas as pd
+from cleanlab_studio.utils.data_enrichment.enrichment_utils import (
+    extract_df_subset,
+    get_compiled_regex_list,
+    get_prompt_outputs,
+    get_regex_match,
+    get_return_values_match,
+)
+
+from cleanlab_studio.studio.studio import Studio
+
+
+def enrich_data(
+    studio: Studio,
+    prompt: str,
+    data: pd.DataFrame,
+    *,
+    regex: Optional[Union[str, re.Pattern, List[re.Pattern]]] = None,
+    return_values: Optional[List[str]] = None,
+    subset_indices: Optional[Union[Tuple[int, int], List[int]]] = (0, 3),
+    column_name_prefix: str = "",
+    **kwargs,
+) -> pd.DataFrame:
+    """
+    This method takes in a Studio client object, prompt template and a DataFrame and enriches the dataframe with the results of the prompting and associated trustworthiness scores.
+
+    Args:
+        prompt: Formatted f-string, that contains both the prompt, and names of columns to embed.
+            **Example:** "Is this a numeric value, answer Yes or No only. Value: {column_name}"
+        regex: One or more expressions will be passed into re.compile or a list of already compiled regular expressions.
+            If a list is proivded, the regexes are applied in order and first succesfull match is returned.
+            **Note:** Regex patterns should each specify exactly 1 group that is the match group using parenthesis like so '.*(<desired match group pattern>)'.
+            **Example:** `r'.*(Bird|[Rr]abbit).*'` will match any string that is the word 'Bird', 'Rabbit' or 'rabbit' into group 1.
```

Comment:
why do you say `match` here? you should instead say what the implications are for the returned metadata. The user's goal here is not to match things with their regex, it is to get a well-formatted metadata column, so write the docstring with that goal in mind

Reply:
can you show separate examples of regexes to extract some part of the raw LLM output, and one to replace some part of the raw LLM output?

as well as an other fundamental examples you can think of (eg. handling punctuation at the end of raw llm outputs which is sometimes there sometimes missing)

Reply:
add sentence at end of this arg's docstring:

This `regex` argument is useful in settings where you are unable to prompt the LLM to generate valid outputs 100% of the time, but can easily transform the raw LLM outputs to be valid through regular expressions that extract or replace parts of the raw output string.